chaincode/src/allowances/checkAllowances.ts

```diff
@@ -111,7 +111,7 @@ export async function checkAllowances(
       allowance.additionalKey === tokenInstanceKey.additionalKey &&
       allowance.instance.isEqualTo(tokenInstanceKey.instance) &&
       allowance.allowanceType === action &&
-      (allowance.expires === 0 || isAllowanceExpired(ctx, allowance))
+      (allowance.expires === 0 || !isAllowanceExpired(ctx, allowance))
```

Comment:
I notice the `isAllowanceExpired()` function returns false if expires is zero, so I suppose this line could just be `!isAllowanceExpired(ctx, allowance)` without checking the zero case here.

Reply:
Good call, simpler is better. I pushed a small update